src/frontend/screens/ObservationMetadata.tsx

```diff
@@ -231,12 +231,38 @@ export const ObservationMetadata: NativeNavigationComponent<
     },
   );
 
+  function handleNumberInString(raw: string, decimals: number): string {
+    return raw.replace(/-?\d+(?:\.\d+)?/, match => {
+      const n = Number(match);
+      if (!Number.isFinite(n)) return match;
+      return n.toFixed(decimals);
+    });
+  }
+
+  function formatValueForShare(key: string, raw: number | string): string {
+    const decimals =
+      key === 'latitude' || key === 'longitude' ? 5 : key === 'speed' ? 2 : 0;
+
+    if (typeof raw === 'number') return raw.toFixed(decimals);
+    return handleNumberInString(raw, decimals);
+  }
```

Comment:
this approach is a little fragile and requires implicit developer knowledge. `key` being any string, and transforming the value to zero decimals unless it is lat, lon, or speed is highly specific, and the introduction of any new values will automatically round that new value to 0, which may not be the intended purpose.

I think a better approach would be to update the values in the `listData` array directly using Number.toFixed(). This also would mean you would not have to use `handleNumberInString` as all the values are numbers before they are turned into strings

I know this would update the value in comapeo (before it is shared). But i think that is an ok compromise